src/particles/beam/BeamParticleContainer.cpp

```diff
@@ -133,6 +133,9 @@ BeamParticleContainer::InitData (const amrex::Geometry& geom)
         getWithParser(pp, "num_particles", m_num_particles);
         bool charge_is_specified = queryWithParser(pp, "total_charge", m_total_charge);
         bool peak_density_is_specified = queryWithParser(pp, "density", m_density);
+        if (charge_is_specified) AMREX_ALWAYS_ASSERT_WITH_MESSAGE( Hipace::m_normalized_units == 0,
+            "The option 'beam.total_charge' is only valid in SI units."
+            "Please set 'hipace.normalized_units = 0'");
```

Comment:
I am not sure this is the most natural course of action for a user writing an input file in normalized units, so maybe this could make the message a bit more helpful?
```suggestion
            "Please either specify the peak density with '<beam name>.density', "
            "or set 'hipace.normalized_units = 0' to run in SI units, and update the input file accordingly.");
```